python/lancedb/table.py

```diff
@@ -92,12 +98,13 @@ def create_index(self, num_partitions=256, num_sub_vectors=96):
             The number of PQ sub-vectors to use when creating the index.
             Default is 96.
         """
-        return self._dataset.create_index(
```

Comment:
before the PR, this `create_index` does not return new dataset? if so, is it a bug from `lance` side?

Reply:
it does, but we don't necessarily want it to return a lance dataset because that's implementation detail from the db's perspective